lib/models/Compare_Crypto.rb

```diff
@@ -44,15 +50,23 @@ def get_fiat_input
         return fiat_input
     end
     
-    def set_crypto_1(crypto_input,fiat_input)
-        @crypto_1 = CRYPTOAPI.get_crypto_price(crypto_input,fiat_input)
-        @crypto_1_historical = CRYPTOAPI.get_historical_data(crypto_input,fiat_input)
-        puts @crypto_1,@crypto_1_historical
+    def set_crypto(crypto_input,fiat_input)
+        @crypto << [CRYPTOAPI.get_crypto_price(crypto_input,fiat_input),CRYPTOAPI.get_historical_data(crypto_input,fiat_input)]
     end
 
-    def set_crypto_2(crypto_input,fiat_input)
-        @crypto_2 = CRYPTOAPI.get_crypto_price(crypto_input,fiat_input)
-        @crypto_2_historical = CRYPTOAPI.get_historical_data(crypto_input,fiat_input)
-        puts @crypto_2,@crypto_2_historical
+    def calculate_historical_change(crypto_input,fiat_input,crypto,crypto_historical)
+        current_price = crypto[2]
+        price_change_14d = (((current_price - crypto_historical[0])/crypto_historical[0])*100).round(3)
+        price_change_7d = (((current_price - crypto_historical[1])/crypto_historical[1])*100).round(3)
+        set_table(crypto_input,fiat_input,price_change_7d,price_change_14d,crypto,crypto_historical)
+    end
+
+    def set_table(crypto_input,fiat_input,price_change_7d,price_change_14d,crypto,crypto_historical)
+        values = crypto
+        historical_values = crypto_historical
+        values[5].positive? == true ? values [5] = "▲ +#{values[5]}%".to_s.green : values [5] = "▼ #{values[5]}%".to_s.red
```

Comment:
Nice use of a ternary, but what is it you're trying to do to values[5] here? We shouldn't change the contents of an array that has been passed in when possible.

`.positive?` already returns a boolean value (the ? makes it operate on the value its being called on), so you don't then need to do the true/false check after

An alternative would be to pass this check into a value and then return/display that however you want after
For example:
`
display = values[5].positive? ?  ▲ +#{values[5]}%".to_s.green" : "▼ #{values[5]}%".to_s.red
`